docs/prompts.md

```diff
@@ -0,0 +1,103 @@
+# Prompts
+
+Prompt manipulation is the primary way that you, as an app developer, influence
+the output of generative AI models. For example, when using LLMs, you can craft
+prompts that influence the tone, format, length, and other characteristics of
+the models’ responses.
+
+Genkit is designed around the premise that _prompts are code_. You write and
+maintain your prompts in source files, track changes to them using the same version
+control system that you use for your code, and you deploy them along with the code
+that calls your generative AI models.
+
+Most developers will find that the included [Dotprompt](./dotprompt.md) library
+meets their needs for working with prompts in Genkit. However, alternative
+approaches are also supported by working with prompts directly.
+
+## Defining prompts
+
+Genkit's `generate()` helper function accepts string prompts, and you can
+call models this way for straight-forward use cases.
+
+```ts
+import { generate } from '@genkit-ai/ai';
+
+generate({
+  model: 'googleai/gemini-pro',
+  prompt: 'You are a helpful AI assistant named Walt.',
+});
+```
+
+In most cases, you will need to include some customer provided inputs in your prompt.
+You could define a function to render them like this.
+
+```ts
+function helloPrompt(name: string) {
+  return `You are a helpful AI assistant named Walt. Say hello to ${name}.`;
+}
+
+generate({
+  model: 'googleai/gemini-pro',
+  prompt: helloPrompt('Fred'),
+});
+```
+
+One shortcoming of defining prompts in your code is that testing requires executing
+them as part of a flow. To faciliate more rapid iteration, Genkit provides a facility
+to define your prompts and run them in the Developer UI.
+
+Use the `definePrompt` function to register your prompts with Genkit.
+
+```ts
+import { definePrompt } from '@genkit-ai/ai';
+import z from 'zod';
+
+export const helloPrompt = definePrompt(
+  {
+    name: 'helloPrompt',
+    inputSchema: z.object({ name: z.string() }),
+  },
+  async (input) => {
+    const promptText = `You are a helpful AI assistant named Walt.
+    Say hello to ${input.name}.`;
+
+    return {
+      messages: [{ role: 'user', content: [{ text: promptText }] }],
```

Comment:
optional: call out that you can also render model config options here as well (temperature, etc...).

Reply:
Added.